sql/auth/sql_user.cc

```diff
@@ -1258,6 +1258,17 @@ bool set_and_validate_user_attributes(
     inbuflen = (unsigned)Str->auth.length;
     std::string gen_password;
     if (Str->has_password_generator) {
+#ifdef WITH_WSREP
+      if (WSREP(thd)) {
+        const char *msg = "Percona XtraDB Cluster doesn't allow use of"
+            " RANDOM PASSWORD for CREATE/ALTER USER operation"
+            " while operating in cluster mode";
+        WSREP_ERROR("%s", msg);
+        my_message(ER_UNKNOWN_ERROR, msg, MYF(0));
+        plugin_unlock(0, plugin);
```

Comment:
The other places where we return an error after the plugin lock, I see this is getting set
  what_to_set.m_what = NONE_ATTR;
should we also do this here?

Reply:
Shouldn't be a big deal, because in case of error it is not used, and at the top of this function it is set to NONE_ATR, but anyway, good to stick to convention in this function.
Fixed.

Reply:
Fixed.